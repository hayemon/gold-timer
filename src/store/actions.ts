export { addReward, incrementTotalOverdrives } from './slices/statSlice'
export {
  setInactive,
  setLastValue,
  setOverdriveCount,
  setPressCount,
  setTimerCount,
} from './slices/timerSlice'
