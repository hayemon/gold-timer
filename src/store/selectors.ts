export { selectStat } from './slices/statSlice'
export { selectTimer } from './slices/timerSlice'
