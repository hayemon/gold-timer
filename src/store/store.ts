import {
  combineReducers,
  configureStore,
  PreloadedState,
} from '@reduxjs/toolkit'
import statReducer from './slices/statSlice'
import timerReducer from './slices/timerSlice'

const rootReducer = combineReducers({
  timer: timerReducer,
  stat: statReducer,
})

export const createStore = (preloadedState?: PreloadedState<RootState>) =>
  configureStore({
    reducer: rootReducer,
    preloadedState,
  })

export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof createStore>
export type AppDispatch = AppStore['dispatch']
