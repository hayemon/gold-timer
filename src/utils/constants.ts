export const defaultStep = 1
export const overdriveStep = 2

export const overdriveTimeout = 10000
export const inactiveTimeout = 10000
export const defaultDelay = 1000
