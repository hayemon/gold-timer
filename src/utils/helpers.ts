export const getStep = (overdriveCount: number): number => {
  if (overdriveCount > 0) {
    return 2
  }
  return 1
}
