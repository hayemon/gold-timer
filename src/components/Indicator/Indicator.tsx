import { LoadingOutlined } from '@ant-design/icons'
import { Spin, Typography } from 'antd'
import React from 'react'
import { selectTimer } from 'store/selectors'

const Indicator: React.FC = () => {
  const { overdriveCount, inactive } = selectTimer()

  let className = 'treasure-box'
  if (inactive) {
    className += ' treasure-box__shaded'
  }

  return (
    <div className='indicator-container'>
      {overdriveCount > 0 && (
        <>
          <Typography.Title level={5} type='danger'>
            OVERDRIVE!
          </Typography.Title>
          <Spin
            indicator={
              <LoadingOutlined style={{ fontSize: 80, color: 'danger' }} spin />
            }
          />
          <br />
          <Typography.Title level={3} type='danger'>
            {(overdriveCount / 60).toFixed(1)}
          </Typography.Title>
        </>
      )}
    </div>
  )
}

export default Indicator
