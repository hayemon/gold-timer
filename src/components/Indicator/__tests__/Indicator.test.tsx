import { screen } from '@testing-library/react'
import { setOverdriveCount } from 'store/actions'
import { renderWithProviders } from 'store/testUtils'
import { overdriveStep } from 'utils/constants'
import Indicator from '../Indicator'

describe('Indicator', () => {
  it('renders', () => {
    const { asFragment } = renderWithProviders(<Indicator />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('change overdriveCount', () => {
    const { store } = renderWithProviders(<Indicator />)

    store.dispatch(setOverdriveCount(overdriveStep))
    expect(store.getState().timer.overdriveCount).toBe(overdriveStep)
    screen.getByText('OVERDRIVE!')
    const seconds = (overdriveStep / 60).toFixed(1)
    screen.getByText(seconds)

    store.dispatch(setOverdriveCount(0))
    expect(store.getState().timer.overdriveCount).toBe(0)
  })
})
