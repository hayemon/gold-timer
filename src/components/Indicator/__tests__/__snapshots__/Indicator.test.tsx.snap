// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator renders 1`] = `
<DocumentFragment>
  <div
    class="indicator-container"
  />
</DocumentFragment>
`;
